pkg/services/datasources/models.go

```diff
@@ -70,6 +70,16 @@ type DataSource struct {
 
 	Created time.Time `json:"created,omitempty"`
 	Updated time.Time `json:"updated,omitempty"`
+
+	isSecureSocksDSProxyEnabled *bool `xorm:"-"`
+}
+
+func (ds *DataSource) IsSecureSocksDSProxyEnabled() bool {
+	if ds.isSecureSocksDSProxyEnabled == nil {
+		enabled := ds.JsonData != nil && ds.JsonData.Get("enableSecureSocksProxy").MustBool(false)
+		ds.isSecureSocksDSProxyEnabled = &enabled
+	}
+	return *ds.isSecureSocksDSProxyEnabled
```

Comment:
I assume that we don't also need to confirm that PDC is actually enabled/configured for the instance too? (Or maybe that logic is determined separately and then combined with this information wherever the validation is done)

Reply:
that's a really good point, I guess we do want to check that
I guess when PDC is not enabled in the instance we fallback into not using PDC regardless of the datasource configuration right?

Reply:
Exactly. Without the configuration, no PDC connection will be setup. 

So I suppose the easiest thing would be adding `*cfg.Setting` here and checking `cfg.ProxySettings.Enabled` as well.